Jenkinsfile

```diff
@@ -0,0 +1,197 @@
+#!groovy
+
+/**
+
+    Requirements for this Jenkinsfile:
+    - JDK8 with the id "jdk8"
+    - Maven 3.2.5 with the id "Maven 3.2.5"
+    - Plugin "Pipeline Utility Steps"
+    - Plugin "Xvfb" with an installation called "System"
+
+    The following signatures need to be approved:
+    method hudson.plugins.git.GitSCM getUserRemoteConfigs
+    method hudson.plugins.git.UserRemoteConfig getUrl
+    method java.util.Collection addAll java.util.Collection
+    staticMethod org.codehaus.groovy.runtime.DefaultGroovyMethods first java.util.List
+    staticMethod org.codehaus.groovy.runtime.DefaultGroovyMethods stripIndent java.lang.String
+*/
+
+nodeWithProperWorkspace {
+    stage 'Checkout'
+    checkout scm
+    if (isMaster()) {
+        // git by default checks out detached, we need a local branch
+        sh "git checkout $env.BRANCH_NAME" // workaround for https://issues.jenkins-ci.org/browse/JENKINS-31924
+        sh 'git fetch --prune origin +refs/tags/*:refs/tags/*' // delete all local tags
+        sh "git reset --hard origin/master"
+        if (isTag()) {
+            // Workaround: we don't want infinite releases.
+            echo "Aborting build as the current commit on master is already tagged."
+            return
+        }
+    }
+    sh "git clean -ffdx"
+
+    def preReleaseVersion = getCurrentVersion()
+    if (isMaster()) {
+        prepareRelease()
+    }
+
+    stage 'Build target platform'
+    withMavenEnv {
+        mvn 'clean install -f "releng/org.testeditor.releng.target/pom.xml"'
+    }
+
+    stage (isMaster() ? 'Build and deploy' : 'Build')
+    withMavenEnv(["MAVEN_OPTS=-Xms512m -Xmx2g"]) {
+        // workaround for now to speed-up the build: only build the product on develop and master
+        def buildProduct = env.BRANCH_NAME == 'develop' || isMaster() ? '-Pproduct' : ''
```

Comment:
We only need this here, would introduce it as soon as we've got another occurrence.